src/fundus/scraping/url.py

```diff
@@ -0,0 +1,130 @@
+                an upper bound and not necessarily the actual number of
+                URLs. If set to None, the source will be exhausted until
+                StopIteration is hit. Defaults to None.
+
+        Yields:
+            str: The next URL.
+        """
+        return itertools.islice(self, max_urls)
+
+
+@dataclass
+class RSSFeed(URLSource):
+    def __iter__(self) -> Iterator[str]:
+        session = session_handler.get_session()
+        response = session.get(self.url, headers=self._request_header)
+        html = response.text
+        rss_feed = feedparser.parse(html)
+        if exception := rss_feed.get("bozo_exception"):
+            basic_logger.warning(f"Warning! Couldn't parse rss feed '{self.url}' because of {exception}")
+            return
+        else:
+            for url in (entry["link"] for entry in rss_feed["entries"]):
+                yield url
+
+
+@dataclass
+class Sitemap(URLSource):
+    recursive: bool = True
+    reverse: bool = False
+    sitemap_filter: URLFilter = lambda url: not bool(url)
+
+    _decompressor: ClassVar[_ArchiveDecompressor] = _ArchiveDecompressor()
+    _sitemap_selector: ClassVar[XPath] = CSSSelector("sitemap > loc")
+    _url_selector: ClassVar[XPath] = CSSSelector("url > loc")
+
+    def __iter__(self) -> Iterator[str]:
+        def yield_recursive(sitemap_url: str) -> Iterator[str]:
+            session = session_handler.get_session()
+            if not validators.url(sitemap_url):
+                basic_logger.info(f"Skipped sitemap '{sitemap_url}' because the URL is malformed")
+            try:
+                response = session.get(url=sitemap_url, headers=self._request_header)
+            except (HTTPError, ConnectionError) as error:
+                basic_logger.warning(f"Warning! Couldn't reach sitemap '{sitemap_url}' because of {error}")
+                return
+            content = response.content
+            if (content_type := response.headers["content-type"]) in self._decompressor.supported_file_formats:
+                content = self._decompressor.decompress(content, content_type)
+            if not content:
+                basic_logger.warning(f"Warning! Empty sitemap at '{sitemap_url}'")
+                return
+            tree = lxml.html.fromstring(content)
+            urls = [node.text_content() for node in self._url_selector(tree)]
+            if urls:
+                for new_url in reversed(urls) if self.reverse else urls:
+                    yield new_url
+            elif self.recursive:
+                sitemap_locs = [node.text_content() for node in self._sitemap_selector(tree)]
+                filtered_locs = list(filter(inverse(self.sitemap_filter), sitemap_locs))
+                for loc in reversed(filtered_locs) if self.reverse else filtered_locs:
+                    yield from yield_recursive(loc)
+
+        yield from yield_recursive(self.url)
+
+
+@dataclass
+class NewsMap(Sitemap):
```

Comment:
Is there a need to have this class? To me it seems that it is treated as any other `SiteMap`

Reply:
Yes, this is necessary in order to be able to differentiate between `Sitemap` and `NewsMap`